src/activities/Elsa.Activities.Webhooks/ActivityTypes/WebhookActivityTypeProvider.cs

```diff
@@ -0,0 +1,99 @@
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Elsa.Activities.Http;
+using Elsa.Activities.Webhooks.Models;
+using Elsa.Activities.Webhooks.Persistence;
+using Elsa.ActivityProviders;
+using Elsa.Metadata;
+using Elsa.Persistence.Specifications;
+using Elsa.Services;
+using Elsa.Services.Models;
+using Newtonsoft.Json.Linq;
+
+namespace Elsa.Activities.Webhooks.ActivityTypes
+{
+    public class WebhookActivityTypeProvider : IActivityTypeProvider
+    {
+        private const string WebhookActivityCategory = "Webhooks";
+
+        private readonly IWebhookDefinitionStore _webhookDefinitionStore;
+        private readonly IActivityActivator _activityActivator;
+
+        public WebhookActivityTypeProvider(
+            IWebhookDefinitionStore webhookDefinitionStore,
+            IActivityActivator activityActivator)
+        {
+            _webhookDefinitionStore = webhookDefinitionStore;
+            _activityActivator = activityActivator;
+        }
+
+        public async ValueTask<IEnumerable<ActivityType>> GetActivityTypesAsync(CancellationToken cancellationToken = default)
+        {
+            var specification = Specification<WebhookDefinition>.All;
+            var definitions = await _webhookDefinitionStore.FindManyAsync(specification, cancellationToken: cancellationToken);
+
+            var activityTypes = new List<ActivityType>();
+            foreach (var definition in definitions)
+            {
+                var activity = CreateWebhookActivityType(definition);
+                activityTypes.Add(activity);
+            }
+
+            return activityTypes;
+        }
+
+        private ActivityType CreateWebhookActivityType(WebhookDefinition webhook)
+        {
+            var typeName = webhook.Name;
+            var displayName = webhook.Name;
+
+            var descriptor = new ActivityDescriptor
+            {
+                Type = typeName,
+                DisplayName = displayName,
+                Category = WebhookActivityCategory,
+                Outcomes = new[] { OutcomeNames.Done },
+                Traits = ActivityTraits.Trigger,
+                Properties = new[]
+                {
+                    new ActivityPropertyDescriptor(
+                        "RequestMethod",
+                        "select",
+                        "Request Method",
+                        "Specify what request method this webhook should handle. Leave empty to handle both GET and POST requests",
+                        JObject.FromObject(new { Options = new[] { "", "GET", "POST" } }))
+                }
+            };
+
+            async ValueTask<IActivity> ActivateActivityAsync(ActivityExecutionContext context, WebhookDefinition webhook)
+            {
+                var activity = await _activityActivator.ActivateActivityAsync(context, typeof(HttpEndpoint));
+
+                if (activity is HttpEndpoint httpEndpointActivity)
```

Comment:
It appears to me that this check is unnecessary - `_activityActivator.ActivateActivityAsync(context, typeof(HttpEndpoint))` will always return an activity of type `HttpEndpoint` (unless there's a bug, in which case we want to fail hard).